tests/test_TFNetworkRecLayer.py

```diff
@@ -4609,6 +4609,26 @@ def make_extern_data():
     assert "encoder_int" in cell.input_layers_moved_out
 
 
+def test_reclayer_optimize_out_cumsum_step_by_step():
+  from returnn.tf.util.data import BatchDim, DimensionTag
+  time_dim = DimensionTag(kind=DimensionTag.Types.Spatial, description="time")
+  feat_dim = DimensionTag(kind=DimensionTag.Types.Feature, description="feat", dimension=11)
+  check_reclayer_optimize_out(
+    subnet_layer_dict={
+      "class": "cumsum", "axis": time_dim, "out_shape": {BatchDim, feat_dim}, "n_out": 11},
```

Comment:
Why do you need `n_out`?

This looks inconsistent. If at all, I would use `out_dim` instead of `n_out`. But then I don't think you need it at all here, right?

Reply:
I need `n_in = n_out` in `check_reclayer_optimize_out`, which has this logic currently:
```
  n_out = subnet_layer_dict.get("n_out", 17)
  # ...
  rec_layer_dict = {
    "class": "rec", ... "n_out": n_out
  }
```
I could also set the my input dim (my `feat_dim`) to 17 to match this default.

Reply:
Maybe it's nicer how I have it now?

Reply:
Ah this is just because of `check_reclayer_optimize_out`. Well yea, it doesn't really matter much then.